src/components/cylc/Task.vue

```diff
@@ -158,10 +158,16 @@
 <script>
     export default {
         name: 'Task',
-        data: function () {
-            return {
-                'status': 'running',
-                'progress': 25
+        props: {
+            status: {
+                type: String,
+                required: true,
+                default: 'waiting'
```

Comment:
*Purely out of academic interest* is there a way of using an enum from this, I think we can extract the allowed values using introspection on GraphQL.

Reply:
That's in my browser history from yesterday. Something like 'vue enum props component'. Unfortunately there is no clear solution. Some people recommended `type: Function` or `Object` if I recall correctly, then you would have a bit more of control.

A good improvement. Do you think we should create an issue for this, or add a TODO here?

Reply:
We can leave this as is, a nice improvement but not an important one.